File: docs/mexc/mexc-api-sdk-main/mexc-api-sdk-main/packages/nodejs/tests/mexc_bypass.test.js
import { MexcBypass } from '../src/MexcBypass.js';

const config = {
  apiKey: 'YOUR_API_KEY',
  isTestnet: true,
  proxyUrl: null,
  testSymbol: 'BTC_USDT'
};

function roundToStep(value, step) {
  const inverse = 1.0 / step;
  return Math.floor(value * inverse + 0.5) / inverse;
}

async function runTests() {
  const mexc = new MexcBypass(config.apiKey, config.isTestnet, config.proxyUrl);

  try {
    console.log('=== Starting MexcBypass Tests ===');

    console.log('\n1. Testing server time:');
    const serverTime = await mexc.getServerTime();
    console.log('Server Time:', serverTime);

    console.log('\n2. Testing user info:');
    const userInfo = await mexc.getUserInfo();
    console.log('User Info:', userInfo);

    console.log('\n3. Testing balance info:');
    const assets = await mexc.getFuturesAssets();
    console.log('Futures Assets:', assets);

    console.log('\n4. Testing contract info:');
    const contract = await mexc.getFuturesContracts({ symbol: config.testSymbol });
    console.log('Contract Info:', contract);

    const priceScale = contract.data?.[0]?.priceUnit || 0.1;
    const volScale = contract.data?.[0]?.volScale || 0;

    console.log('\n5. Testing current price:');
    const ticker = await mexc.getFuturesTickers({ symbol: config.testSymbol });
    console.log('Ticker:', ticker);
    const currentPrice = ticker.data?.lastPrice;

    if (!currentPrice) {
      throw new Error('Failed to get current price');
    }

    console.log('\n6. Testing limit order creation:');
    const rawOrderPrice = currentPrice * 0.9;
    const orderPrice = roundToStep(rawOrderPrice, priceScale).toFixed(volScale);
    const createOrderRes = await mexc.createFuturesOrder({
      symbol: config.testSymbol,
      price: orderPrice,
      type: 1,
      openType: 2,
      positionMode: 1,
      side: 1,
      vol: 1,
      leverage: 10
    });
    console.log('Create Order Result:', createOrderRes);

    const orderId = createOrderRes.data.orderId;
    if (!orderId) {
      throw new Error('Failed to create order');
    }

    console.log('\n7. Testing order info:');
    const orderInfo = await mexc.getFuturesOrdersById({ ids: orderId });
    console.log('Order Info:', orderInfo);

    console.log('\n8. Testing order cancellation:');
    const cancelRes = await mexc.cancelFuturesOrders({ ids: [orderId] });
    console.log('Cancel Order Result:', cancelRes);

    console.log('\n9. Testing trigger order creation:');
    const rawTriggerPrice = currentPrice * 1.1;
    const triggerPrice = roundToStep(rawTriggerPrice, priceScale).toFixed(volScale);
    const triggerOrderRes = await mexc.createFuturesTriggerOrder({
      symbol: config.testSymbol,
      price: orderPrice,
      vol: 1,
      leverage: 10,
      side: 2,
      openType: 2,
      triggerPrice: triggerPrice,
      triggerType: 1,
      executeCycle: 1,
      orderType: 1,
      trend: 1
    });
    console.log('Trigger Order Result:', triggerOrderRes);

    const triggerOrderId = triggerOrderRes.data;

    if (!triggerOrderId) {
      throw new Error('Failed to create trigger order');
    }

    console.log('\n10. Testing trigger order cancellation:');
    const cancelTriggerRes = await mexc.cancelFuturesTriggerOrders({ ids: [triggerOrderId] });
    console.log('Cancel Trigger Order Result:', cancelTriggerRes);

    console.log('\n11. Testing open positions:');
    const positions = await mexc.getFuturesOpenPositions({ symbol: config.testSymbol });
    console.log('Open Positions:', positions);

    console.log('\n12. Testing leverage change:');
    if (positions.data && positions.data.length > 0) {
      const positionId = positions.data[0].positionId;
      const changeLeverageRes = await mexc.changeFuturesPositionLeverage({
        positionId: positionId,
        leverage: 10 + 1,
        openType: 2,
        symbol: config.testSymbol,
        positionType: 1
      });
      console.log('Change Leverage Result:', changeLeverageRes);
    } else {
      console.log('No open positions to change leverage');
    }

    console.log('\n=== All tests completed successfully ===');
  } catch (error) {
    console.error('\n=== Error during testing ===');
    console.error('Error:', error.message);
    console.error('Stack:', error.stack);
  }
}

runTests();

File: docs/mexc/mexc-api-sdk-main/mexc-api-sdk-main/packages/python/tests/mexc_bypass.test.py
import asyncio
from src.MexcBypass import MexcBypass

async def test_mexc_bypass():
    async with MexcBypass(
        api_key="your_mexc_token",
        is_testnet=True,
        proxy_url=None
    ) as client:
        

        print("\n1. Testing server time:")
        time_resp = await client.get_server_time()
        print(time_resp)
        

        print("\n2. Testing user info:")
        user_info = await client.get_user_info()
        print(user_info)
        

        print("\n3. Testing futures assets:")
        assets = await client.get_futures_assets()
        print(assets)
        

        print("\n4. Testing futures contracts:")
        contracts = await client.get_futures_contracts({"symbol": "BTC_USDT"})
        print(contracts)
        

        print("\n5. Testing tickers:")
        tickers = await client.get_futures_tickers({"symbol": "BTC_USDT"})
        print(tickers)
        

        print("\n6. Testing order creation:")
        order = await client.create_futures_order({
            "symbol": "BTC_USDT",
            "price": "30000",
            "type": "1",
            "open_type": "1",
            "position_mode": "1",
            "side": "1",
            "vol": "1",
            "leverage": "10"
        })
        print("Order created:", order)
        
        order_id = order.get('data', {}).get('order_id') if order.get('data') else None
        

        if order_id:
            print("\n7. Testing get order by ID:")
            order_info = await client.get_futures_orders_by_id({"ids": str(order_id)})
            print(order_info)
            
            print("\n8. Testing order cancellation:")
            cancel_resp = await client.cancel_futures_orders({"ids": [order_id]})
            print(cancel_resp)
        

        print("\n9. Testing open positions:")
        positions = await client.get_futures_open_positions({"symbol": "BTC_USDT"})
        print(positions)
        

        print("\n10. Testing trigger order creation:")
        trigger_order = await client.create_futures_trigger_order({
            "symbol": "BTC_USDT",
            "price": "25000",
            "vol": "1",
            "leverage": "10",
            "side": "1",
            "open_type": "1",
            "trigger_price": "28000",
            "trigger_type": "1",
            "execute_cycle": "1",
            "order_type": "1",
            "trend": "1"
        })
        print("Trigger order created:", trigger_order)
        
        trigger_order_id = trigger_order.get('data')
        

        print("\n11. Testing trigger orders list:")
        trigger_orders = await client.get_futures_trigger_orders({"symbol": "BTC_USDT"})
        print(trigger_orders)
        

        if trigger_order_id:
            print("\n12. Testing trigger order cancellation:")
            cancel_trigger = await client.cancel_futures_trigger_orders({"ids": trigger_order_id})
            print(cancel_trigger)
        

        print("\n13. Testing leverage change:")
        leverage_resp = await client.change_futures_position_leverage({
            "symbol": "BTC_USDT",
            "leverage": "20",
            "open_type": "1"
        })
        print(leverage_resp)
        

        print("\n14. Testing close all positions:")
        close_all = await client.close_all_futures_positions()
        print(close_all)
        

        print("\n15. Testing orders history:")
        history = await client.get_futures_orders_history({"symbol": "BTC_USDT"})
        print(history)

if __name__ == "__main__":
    asyncio.run(test_mexc_bypass())